### ISSUE TITLE
[BUG] [Feature/Agent_framework] Creating a connector without a request body gives 500 error when using it

### ISSUE THREAD

**Issue by dbwiddis** [opensearch-project/ml-commons#1738]
**What is the bug?**

The create connector API includes a "request_body" field in its actions.  If this element is omitted, the connector still creates successfully, but when it is used, the lack of request body results in a null `payload` variable, which fails in making the HTTP connection.

**How can one reproduce the bug?**
1. Create a connector using `_plugins/_ml/connectors/_create` but in the `"actions"` field, leave the "request_body" parameter blank.
2. Create a model with that connector
3. Use the predict API with that model to ask a question
4. Get a 500 response:
```json
{
    "error": {
        "root_cause": [
            {
                "type": "m_l_exception",
                "reason": "Failed to create http request for remote model"
            }
        ],
        "type": "m_l_exception",
        "reason": "Failed to create http request for remote model",
        "caused_by": {
            "type": "illegal_argument_exception",
            "reason": "Source string may not be null"
        }
    },
    "status": 500
}
```

**What is the expected behavior?**

1. If "request_body" is a required parameter, I would expect a 400-series error creating the connector, informing me of this fact.
2. If "request_body" is an optional parameter, I would expect a reasonable default to an empty body "{}" prior to attempting to use it in the predict call.  Since the null "request_body" falls back to the "http_body" parameter, the default may be better placed where both of these values are null.

**Do you have any additional context?**

The `HttpConnector` class implements `createPredictPayload()` like this:
```java
    @Override
    public  <T> T createPredictPayload(Map<String, String> parameters) {
        Optional<ConnectorAction> predictAction = findPredictAction();
        if (predictAction.isPresent() && predictAction.get().getRequestBody() != null) {
            String payload = predictAction.get().getRequestBody();
            payload = fillNullParameters(parameters, payload);
            StringSubstitutor substitutor = new StringSubstitutor(parameters, "${parameters.", "}");
            payload = substitutor.replace(payload);

            if (!isJson(payload)) {
                throw new IllegalArgumentException("Invalid JSON in payload");
            }
            return (T) payload;
        }
        return (T) parameters.get("http_body");
    }
```
In the case "request_body" is omitted, the conditional fails on `predictAction.get().getRequestBody() != null`, defaulting to return the value of the parameter "http_body".  If that parameter is also empty, the method returns null.

That null return is later processed in `RemoteConnectorExecutor` in `preparePayloadAndInvokeRemoteModel()` here:
```java
        String payload = connector.createPredictPayload(parameters);
        connector.validatePayload(payload);
        invokeRemoteModel(mlInput, parameters, payload, tensorOutputs);
```
The `validatePayload()` method does not fail on a null payload, and the `invoke...()` method throws an `InvalidArgumentException` on the null payload.

**Suggested Fix**
Option 1: change the last line of `createPredictPayload()` to:
```java
return (T) parameters.getOrDefault("http_body", "{}");
```

Option 2: cause `connector.validatePayload()` to fail on null, returning an appropriate 400-series error message about the need for either the connector request_body or http_body to be non-null.

**Comment by mingshl** [comment]
@Zhangxunmt is this issue being addressed? please post update, thank you!

**Comment by Zhangxunmt** [comment]
Added in the 2.16 release plan.

**Comment by Zhangxunmt** [comment]
This bug is already fixed in 2.14, which returns 400 error code with illegal_argument_exception when the payload is null. https://github.com/opensearch-project/ml-commons/blob/main/ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/ConnectorUtils.java#L289.

Verified the new response changed to:
{
  "error": {
    "root_cause": [
      {
        "type": "illegal_argument_exception",
        "reason": "Content length is 0. Aborting request to remote model"
      }
    ],
    "type": "illegal_argument_exception",
    "reason": "Content length is 0. Aborting request to remote model"
  },
  "status": 400
}


Note: the change from 
`return (T) parameters.get("http_body"); `
to 
`return (T) parameters.getOrDefault("http_body", "{}");
`
will not work in some cases when the connector is configured to make a GET request with Null response. For example, I just verified that in the OpenAI GetBatchStatus API, the request body is required to be null. https://platform.openai.com/docs/guides/batch/getting-started.

**Comment by Zhangxunmt** [comment]
Closing this issue as it's fixed already. Please feel free to reopen is there's more concerns.
